backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/OutputGenerator.java

```diff
@@ -99,10 +97,23 @@ public VelocityContext getConfiguredVelocityContext() {
     @NotNull
     protected SortedMap<String, LicenseInfoParsingResult> getSortedLicenseInfos(Collection<LicenseInfoParsingResult> projectLicenseInfoResults) {
         Map<String, LicenseInfoParsingResult> licenseInfos = projectLicenseInfoResults.stream()
-                .collect(Collectors.toMap(this::getComponentLongName, li -> li, (li1, li2) -> li1));
+                .collect(Collectors.toMap(this::getComponentLongName, li -> li, this::mergeLicenseInfoParsingResults));
         return sortStringKeyedMap(licenseInfos);
     }
 
+    @NotNull
+    protected LicenseInfoParsingResult mergeLicenseInfoParsingResults(LicenseInfoParsingResult r1, LicenseInfoParsingResult r2){
+        if (r1.getStatus() != LicenseInfoRequestStatus.SUCCESS || r2.getStatus() != LicenseInfoRequestStatus.SUCCESS ||
+                !getComponentLongName(r1).equals(getComponentLongName(r2))){
+            throw new IllegalArgumentException("Only successful parsing results for the same release can be merged");
+        }
+        LicenseInfoParsingResult r = new LicenseInfoParsingResult(r1);
```

Comment:
Is this equivalent to `r1.deepCopy()` ?
Or are the inner objects still the same, i.e. does the following code also change the lists in `r1`?

Reply:
Yes, it is equivalent. `deepCopy()` is generated by Thrift and does exactly one thing: `    return new LicenseInfoParsingResult(this);`